commons-mongo/jvm/src/main/scala/com/avsystem/commons/mongo/BsonInputOutput.scala

```diff
@@ -54,13 +116,15 @@ trait BsonOutput extends Any with OutputAndSimpleOutput {
     typeMarker match {
       case ObjectIdMarker => writeObjectId(value); true
       case Decimal128Marker => writeDecimal128(value); true
+      case BsonValueMarker => writeBsonValue(value); true
       case _ => false
     }
 }
 
 object BsonOutput {
   def bigDecimalBytes(bigDecimal: BigDecimal): Array[Byte] = {
     val unscaledBytes = bigDecimal.bigDecimal.unscaledValue.toByteArray
-    ByteBuffer.allocate(unscaledBytes.length + Integer.BYTES).put(unscaledBytes).putInt(bigDecimal.scale).array
+    ByteBuffer.allocate(unscaledBytes.length + Integer.BYTES)
+      .put(unscaledBytes).putInt(bigDecimal.scale).array
```

Comment:
Not a part of this change, but sad that we have to create a byte[] array just to copy it to a slightly larger one. I assume this intermediate step could be avoided

Reply:
How?

I can only imagine it if `Input` & `Output` used some kind of `ByteString` instead of raw array but that would be a heavier and breaking change (also, likely to introduce external deps).

Reply:
Copy and modify `toByteArray` logic, use own array and `ByteBuffer.wrap` it if you need to use it as such.

Reply:
Is it ok to not give a damn about this in this review?

Reply:
Your code didn't make anything worse for sure, so that's not a blocker on my side.